src/deadline/cinema4d_adaptor/Cinema4DClient/cinema4d_handler.py

```diff
@@ -50,6 +50,15 @@ def __init__(self, map_path: Callable[[str], str]) -> None:
         self.take = "Main"
         self.map_path = map_path
 
+    def _remap_assets(self):
+        asset_list = []
+        c4d.documents.GetAllAssetsNew(self.doc, allowDialogs=False, lastPath="", assetList=asset_list)
+        for asset in asset_list:
+            asset_owner = asset.get("owner")
+            asset_param_id = asset.get("paramId")
+            if asset_owner and asset_param_id:
+                asset_owner[asset.get("paramId")] = self.map_path(asset.get("filename"))
```

Comment:
I'm trying to better understand this code. How does setting the `asset_owner` variable to a path with applied path mapping rules fix the issue? I don't see `asset_owner` referenced elsewhere else in the code.

Reply:
Hmm, nevermind looks like the object from asset.get("owner") has its paramId modified and that fixes the issue.

Reply:
So the `asset` in this case is a dictionary containing all the information about the asset as it is used in the scene.
What we care most about are the `owner` (what node/shader/material/etc is using this asset), the `paramId` (which property in the owner that contains this asset), and the `filename` (The filename path on disk).

So `asset.get("paramId")` returns the attribute on the owner we need to modify the path of.
In most cases it seems like it will be `c4d.BITMAPSHADER_FILENAME` (for image files) but their documentation seems to want us to use the paramId, probably because there are cases where our asset is not an image file, or if a different node references the asset and doesn't store it in `c4d.BITMAPSHADER_FILENAME`.

Reply:
Thanks for the detailed explanation.